src/backend/distributed/commands/multi_copy.c

```diff
@@ -45,94 +45,37 @@
  */
 
 #include "postgres.h"
-#include "fmgr.h"
-#include "funcapi.h"
 #include "libpq-fe.h"
 #include "miscadmin.h"
-#include "plpgsql.h"
 
-#include <arpa/inet.h>
-#include <netinet/in.h>
-#include <string.h>
-
-#include "access/heapam.h"
 #include "access/htup_details.h"
 #include "access/htup.h"
-#include "access/nbtree.h"
 #include "access/sdir.h"
-#include "access/tupdesc.h"
-#include "access/xact.h"
 #include "catalog/namespace.h"
-#include "catalog/pg_class.h"
 #include "catalog/pg_type.h"
-#include "catalog/pg_am.h"
-#include "catalog/pg_collation.h"
-#include "commands/extension.h"
 #include "commands/copy.h"
 #include "commands/defrem.h"
-#include "distributed/citus_ruleutils.h"
-#include "distributed/commit_protocol.h"
-#include "distributed/connection_cache.h"
-#include "distributed/listutils.h"
-#include "distributed/metadata_cache.h"
-#include "distributed/master_metadata_utility.h"
 #include "distributed/master_protocol.h"
 #include "distributed/metadata_cache.h"
 #include "distributed/multi_copy.h"
 #include "distributed/multi_physical_planner.h"
 #include "distributed/multi_shard_transaction.h"
-#include "distributed/pg_dist_partition.h"
+#include "distributed/placement_connection.h"
+#include "distributed/remote_commands.h"
 #include "distributed/resource_lock.h"
-#include "distributed/shardinterval_utils.h"
-#include "distributed/worker_protocol.h"
-#include "executor/execdesc.h"
 #include "executor/executor.h"
-#include "executor/instrument.h"
-#include "executor/tuptable.h"
-#include "lib/stringinfo.h"
-#include "nodes/execnodes.h"
-#include "nodes/makefuncs.h"
-#include "nodes/memnodes.h"
-#include "nodes/nodeFuncs.h"
-#include "nodes/nodes.h"
-#include "nodes/params.h"
-#include "nodes/parsenodes.h"
-#include "nodes/pg_list.h"
-#include "nodes/plannodes.h"
-#include "nodes/primnodes.h"
-#include "optimizer/clauses.h"
-#include "optimizer/cost.h"
-#include "optimizer/planner.h"
-#include "optimizer/var.h"
-#include "parser/parser.h"
-#include "parser/analyze.h"
-#include "parser/parse_node.h"
-#include "parser/parsetree.h"
-#include "parser/parse_type.h"
-#include "storage/lock.h"
-#include "tcop/dest.h"
-#include "tcop/tcopprot.h"
-#include "tcop/utility.h"
 #include "tsearch/ts_locale.h"
 #include "utils/builtins.h"
-#include "utils/elog.h"
-#include "utils/errcodes.h"
-#include "utils/guc.h"
 #include "utils/lsyscache.h"
-#include "utils/typcache.h"
-#include "utils/palloc.h"
 #include "utils/rel.h"
-#include "utils/relcache.h"
-#include "utils/snapmgr.h"
-#include "utils/tuplestore.h"
 #include "utils/memutils.h"
 
```

Comment:
After api changes I felt like some of the includes should be unnecessary and ended up finding a lot.